controllers/doguUpgradeManager.go

```diff
@@ -0,0 +1,159 @@
+		return err
+	}
+
+	upgradeDoguName := doguResource.Spec.Name
+	upgradeDoguVersion := doguResource.Spec.Version
+
+	fromDogu, toDogu, developmentDoguMap, err := dum.getDogusForUpgrade(ctx, doguResource)
+	if err != nil {
+		return err
+	}
+
+	dum.normalEvent(doguResource, "Checking premises...")
+	err = dum.premisesChecker.Check(ctx, doguResource, fromDogu, toDogu)
+	if err != nil {
+		dum.errorEventf(doguResource, ErrorOnFailedPremisesUpgradeEventReason, "Checking premises failed: %s", err)
+		return fmt.Errorf("dogu upgrade %s:%s failed a premise check: %w", upgradeDoguName, upgradeDoguVersion, err)
+	}
+
+	dum.normalEventf(doguResource, "Executing upgrade from %s to %s...", fromDogu.Version, toDogu.Version)
+	err = dum.upgradeExecutor.Upgrade(ctx, doguResource, toDogu)
+	if err != nil {
+		dum.errorEventf(doguResource, ErrorOnFailedUpgradeEventReason, "Error during upgrade: %s", err)
+		return fmt.Errorf("dogu upgrade %s:%s failed: %w", upgradeDoguName, upgradeDoguVersion, err)
+	}
+	// note: there won't exist a purgeOldContainerImage step: that is the subject of Kubernetes's cluster configuration
+
+	err = doguResource.ChangeState(ctx, dum.client, k8sv1.DoguStatusInstalled)
+	if err != nil {
+		return err
+	}
+
+	if developmentDoguMap != nil {
+		err = developmentDoguMap.DeleteFromCluster(ctx, dum.client)
+		if err != nil {
+			// an error during deleting the developmentDoguMap is not critical so we change the dogu state as installed earlier
+			dum.errorEventf(doguResource, ErrorOnFailedUpgradeEventReason, "Error during upgrade: %s", err)
+			return fmt.Errorf("dogu upgrade %s:%s failed: %w", upgradeDoguName, upgradeDoguVersion, err)
+		}
+	}
+
+	return nil
+}
+
+func (dum *doguUpgradeManager) getDogusForUpgrade(ctx context.Context, doguResource *k8sv1.Dogu) (*core.Dogu, *core.Dogu, *k8sv1.DevelopmentDoguMap, error) {
+	fromDogu, err := dum.localDoguFetcher.FetchInstalled(doguResource.Name)
+	if err != nil {
+		dum.errorEventf(doguResource, ErrorOnFailedUpgradeEventReason, "Error getting installed dogu for upgrade: %s", err)
+		return nil, nil, nil, fmt.Errorf("dogu upgrade failed: %w", err)
+	}
+
+	toDogu, developmentDoguMap, err := dum.resourceDoguFetcher.FetchWithResource(ctx, doguResource)
+	if err != nil {
+		dum.errorEventf(doguResource, ErrorOnFailedUpgradeEventReason, "Error getting remote dogu for upgrade: %s", err)
+		return nil, nil, nil, fmt.Errorf("dogu upgrade failed: %w", err)
+	}
+
+	return fromDogu, toDogu, developmentDoguMap, nil
+}
+
+func (dum *doguUpgradeManager) normalEvent(doguResource *k8sv1.Dogu, msg string) {
+	dum.eventRecorder.Event(doguResource, corev1.EventTypeNormal, UpgradeEventReason, msg)
+}
+
+func (dum *doguUpgradeManager) normalEventf(doguResource *k8sv1.Dogu, msg string, msgArg ...interface{}) {
+	dum.eventRecorder.Eventf(doguResource, corev1.EventTypeNormal, UpgradeEventReason, msg, msgArg...)
+}
+
+func (dum *doguUpgradeManager) errorEventf(doguResource *k8sv1.Dogu, reason, msg string, err error) {
```

Comment:
This method is no longer needed after removing the error event calls from the manager. It can also be removed.

- [ ] Done